core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java

```diff
@@ -44,63 +46,129 @@ public interface TransactionCrudOperable {
 
   /**
    * Inserts/Updates an entry to the storage through a transaction with the specified {@link Put}
-   * command.
+   * command. If a condition is specified in the {@link Put} command, and if the condition is not
+   * satisfied or the entry does not exist, it throws {@link UnsatisfiedConditionException}.
    *
    * @param put a {@code Put} command
    * @throws CrudConflictException if the transaction CRUD operation fails due to transient faults
    *     (e.g., a conflict error). You can retry the transaction from the beginning
    * @throws CrudException if the transaction CRUD operation fails due to transient or nontransient
    *     faults. You can try retrying the transaction from the beginning, but the transaction may
    *     still fail if the cause is nontranient
-   * @throws UnsatisfiedConditionException if the mutation condition is not satisfied
+   * @throws UnsatisfiedConditionException if a condition is specified, and if the condition is not
+   *     satisfied or the entry does not exist
+   * @deprecated As of release 3.13.0. Will be removed in release 5.0.0.
    */
+  @Deprecated
   void put(Put put) throws CrudConflictException, CrudException, UnsatisfiedConditionException;
 
   /**
    * Inserts/Updates multiple entries to the storage through a transaction with the specified list
```

Comment:
Since I don't think `Insert` and `Update` are interchangeable, how about if we change this part as follows?:

```suggestion
   * Inserts multiple entries into or updates multiple entries in the underlying storage through a transaction with the specified list
```